juju/client/connection.py

```diff
@@ -357,7 +358,7 @@ async def close(self, to_reconnect: bool = False):
             tasks_need_to_be_gathered.append(self._debug_log_task)
             self._debug_log_task.cancel()
 
-        if self._ws and not self._ws.closed:
+        if self._ws and self._ws.state is not State.CLOSED:
```

Comment:
Looking through websockets code, the library already handles the corner cases and we should close the connection unconditionally.